runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -87,1207 +84,395 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
-		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
-	}
-
-	if mv.TimeDimension == "" && (!isTimeRangeNil(q.TimeRange) || !isTimeRangeNil(q.ComparisonTimeRange)) {
-		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsViewName)
-	}
-
-	// backwards compatibility
-	if q.Filter != nil {
-		if q.Where != nil {
-			return fmt.Errorf("both filter and where is provided")
-		}
-		q.Where = convertFilterToExpression(q.Filter)
-	}
-
 	err = q.calculateMeasuresMeta()
 	if err != nil {
 		return err
 	}
 
-	// comparison toplist
-	if !isTimeRangeNil(q.ComparisonTimeRange) {
-		// execute toplist for base and get dim list
-		// create and add filter and execute comprison toplist
-		// remove strict limits in comp toplist sql
-		if drivers.DialectDruid != olap.Dialect() || q.Exact {
-			return q.executeComparisonToplist(ctx, olap, mv, priority, security)
-		}
-
-		return q.executeDruidApproximateComparison(ctx, olap, mv, priority, security)
-	}
-
-	// general toplist
-	if drivers.DialectDruid != olap.Dialect() || q.Exact {
-		return q.executeToplist(ctx, olap, mv, priority, security)
-	}
-
-	return q.executeDruidApproximateToplist(ctx, olap, mv, priority, security)
-}
-
-// Druid-based `exactify` approach:
-// 1. The first query fetch topN dimensions.
-// 2. The second query fetches topN filtered by the collected dimensions.
-// The dimension filter contrains topN table avoiding approximation in measures (due to mearging multiple topN Druid results from different nodes).
-func (q *MetricsViewComparison) executeDruidApproximateComparison(ctx context.Context, olap drivers.OLAPStore, mv *runtimev1.MetricsViewSpec, priority int, security *runtime.ResolvedMetricsViewSecurity) error {
-	originalMeasures := q.removeNoSortMeasures()
-
-	if q.isBase() || q.isDeltaComparison() {
-		err := q.executeToplist(ctx, olap, mv, priority, security)
-		if err != nil {
-			return err
-		}
-	} else {
-		ttr := q.TimeRange
-		q.TimeRange = q.ComparisonTimeRange
-		err := q.executeToplist(ctx, olap, mv, priority, security)
-		if err != nil {
-			return err
-		}
-
-		q.TimeRange = ttr
-	}
-
-	q.addDimsAsFilter()
-	q.Measures = originalMeasures
-	return q.executeComparisonToplist(ctx, olap, mv, priority, security)
-}
-
-// Druid-based `exactify` approach (see comments above)
-// Optimizations:
-// * the first query fetches only sorted dimensions
-// * the second query isn't run if the topN already less than the limit
-func (q *MetricsViewComparison) executeDruidApproximateToplist(ctx context.Context, olap drivers.OLAPStore, mv *runtimev1.MetricsViewSpec, priority int, security *runtime.ResolvedMetricsViewSecurity) error {
-	originalMeasures := q.Measures
-	if len(q.Measures) >= 5 {
-		originalMeasures = q.removeNoSortMeasures()
-	}
-
-	err := q.executeToplist(ctx, olap, mv, priority, security)
+	// Attempt to route to metricsview executor
```

Comment:
Given we return error if that attempt fails we should remove the word Attempt

Reply:
Fixed